client/src/featureform/providers/online_store.py

```diff
@@ -0,0 +1,28 @@
+class OnlineStore:
+    def create_table(self, name, variant, entity_type):
+        raise NotImplementedError()
+
+    def get_table(self, name, variant):
+        raise NotImplementedError()
+
+    def table_exists(self, name, variant):
+        raise NotImplementedError()
+
+    def is_vector_store(self):
+        raise NotImplementedError()
+
+
+class OnlineTable:
```

Comment:
I wonder if it makes sense to use python's contextlibmanager for these so that we can do something like. Depends on @epps 's usage

```
with PineconeTable() as p:
   ...
```

to do that we can add the __enter__ and __exit__ methods here (https://realpython.com/python-with-statement/#the-with-statement-approach).

Reply:
Ill come back to this, really just need something basic working for now